langserver/handler.go

```diff
@@ -71,19 +67,24 @@ func (h *LangHandler) resetCaches(lock bool) {
 	if lock {
 		h.mu.Lock()
 	}
-	h.cache = map[typecheckKey]*typecheckResult{}
+
 	h.importGraphOnce = sync.Once{}
 	h.importGraph = nil
-	if lock {
-		h.mu.Unlock()
+
+	if h.typecheckCache == nil {
+		h.typecheckCache = newTypecheckCache()
+	} else {
+		h.typecheckCache.Purge()
```

Comment:
Purge exists due to the mutex for the variable. If we just do `h.typecheckCache = newTypecheckCache()` we have a race condition on setting `h.typecheckCache`.

I have future plans to fix the need to protect setting these variables.